Documentation/network/standalone.rst

```diff
@@ -0,0 +1,1610 @@
+
+Direct Server Return (DSR)
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, Cilium's load-balancer implementation operates in SNAT mode. That is,
+when node-external traffic arrives which is destined to a service VIP, then the
+node is redirecting the request to the remote backend on its behalf by performing
+SNAT. This does not require any additional MTU changes. The cost is that replies
+from the backend need to make the extra hop back to the load-balancer node to
+perform the reverse SNAT translation before returning the packet directly
+to the external client. Additionally, the original client IP is not preserved at
+the time the packet reaches the backend. The SNAT mode has been used in the
+Quick-Start example above.
+
+The extra hop on the reply can be avoided through Direct Server Return (DSR) where
+the backend replies directly to the external client. The Cilium Standalone Gateway
+supports IPIP and IP6IP6 encapsulation for DSR such that it can be used as a `drop-in
+replacement <https://cilium.io/blog/2022/04/12/cilium-standalone-L4LB-XDP/>`_ for
+existing setups relying on netfilter/IPVS or dedicated hardware load-balancers with
+IPIP encapsulation support.
+
+For all its benefits, be aware that the DSR mode might have limitations with regards
+to the underlying fabric when run off-prem in cloud provider networks. The SNAT mode
+remains the most straightforward mode to configure and run and benefits from not
+requiring any constraints on the network.
+
+The original Quick-Start example has been slightly modified to run in DSR mode:
+
+.. parsed-literal::
+
+   docker run --name cilium-gateway -td \\
+     -v /sys/fs/bpf:/sys/fs/bpf \\
+     -v /lib/modules:/lib/modules \\
+     --privileged=true \\
+     --network=host \\
+     quay.io/cilium/cilium:|IMAGE_TAG| \\
+     cilium-agent \\
+     --enable-ipv4=true \\
+     --enable-ipv6=true \\
+     --datapath-mode=lb-only \\
+     --bpf-lb-algorithm=maglev \\
+     --bpf-lb-acceleration=native \\
+     --bpf-lb-mode=dsr \\
+     --bpf-lb-dsr-dispatch=ipip \\
+     --devices=bond0
+
+In this case the original packet will be preserved in the inner header, and therefore this
+mechanism preserves the original client IP address all the way to the backend nodes. The
+outer IP header will contain the load-balancer address as a source address and the selected
+backend address as a destination address.
+
+There are two modes for the encapsulation which can be toggled through ``--bpf-lb-dsr-l4-xlate``.
+The default mode is also the more common scenario, that is, ``--bpf-lb-dsr-l4-xlate=frontend``.
+Both ``frontend`` and ``backend`` options determine how the inner packet is L4 DNATed, for example:
+
+.. parsed-literal::
+
+  docker exec cilium-gateway cilium service list
+  ID   Frontend             Service Type   Backend
+  1    192.168.160.3:8080   ExternalIPs    1 => 192.168.0.3:4444
+
+With the default ``--bpf-lb-dsr-l4-xlate=frontend``, the inbound and outbound packet look
+as follows:
```

Comment:
```suggestion
With the default ``--bpf-lb-dsr-l4-xlate=frontend``, the inbound and outbound packet look
as follows in ``cilium monitor`` output:
```

(or you could just put a cilium monitor example command into the snippet)

Reply:
it was actually tcpdump :)